### PR TITLE
CTCTRADERS-2303 Remove unnecessary assets in project

### PR DESCRIPTION
**Pull request by amaalali**
- Refactor SpecBase
- Add helper mixins for tests: MockNunjucksRendererApp, TestWithApplication, TestWithMocking
- Add base test for controllers
- Add base test for views
- Remove normal journey related assets including navigator, session and data caching including models such as UserAnswers
- Remove unused controller endpoints

### CODE REVIEW

File: test/base/TestWithApplication.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  this: TestSuite with FakeApplicationFactory =>
+
+  final override def fakeApplication(): Application =
+    guiceApplicationBuilder.build()
+
+  /**
+    * An overrideable hook that allows for the extension of the [[GuiceApplicationBuilder]] or
+    * if overridden, can wipe way some of the assumption in this builder.
+    *
+    * @return Optional user answers that if defefined, will be passed to the GuiceApplicationBuilder
```
Comment (ericohwotu) [inline test/base/TestWithApplication.scala:34]: comment not valid

File: conf/app.routes [hunk 2]
```diff
@@ -6,10 +6,4 @@ GET        /assets/*file                                controllers.Assets.versi
...
 GET        /language/:lang                              controllers.LanguageSwitchController.switchToLanguage(lang: String)
 
-GET        /this-service-has-been-reset                 controllers.SessionExpiredController.onPageLoad
-
-GET        /check-your-answers                          controllers.CheckYourAnswersController.onPageLoad
-
-GET        /unauthorised                                controllers.UnauthorisedController.onPageLoad
-
-GET        /accessibility-statement                     controllers.AccessibilityController.onPageLoad
+GET        /accessibility-statement                     controllers.AccessibilityController.onPageLoad
```
Comment (ericohwotu) [inline conf/app.routes:9]: is this need for internal service?

File: app/config/Module.scala [hunk 3]
```diff
@@ -17,19 +17,9 @@
 package config
 
 import com.google.inject.AbstractModule
-import controllers.actions._
-import repositories.{DefaultSessionRepository, SessionRepository}
 
 class Module extends AbstractModule {
 
-  override def configure(): Unit = {
+  override def configure(): Unit = ()
```
Comment (ericohwotu) [inline app/config/Module.scala:23]: is this file needed now?

File: app/config/FrontendAppConfig.scala [hunk 4]
```diff
@@ -25,7 +25,8 @@ import play.api.mvc.Call
 @Singleton
 class FrontendAppConfig @Inject() (configuration: Configuration) {
 
-  val welshLanguageSupportEnabled: Boolean = configuration.getOptional[Boolean]("features.welsh-language-support").getOrElse(false)
+  val welshLanguageSupportEnabled: Boolean =
```
Comment (ericohwotu) [inline app/config/FrontendAppConfig.scala:28]: are we supporting welsh on this?

File: conf/app.routes (hunk 2 above)
Comment (amaalali) [reply conf/app.routes:9]: I'm not sure so went with keeping the assets for now. I could delete it and bring it back if needed

File: app/config/Module.scala (hunk 3 above)
Comment (amaalali) [reply app/config/Module.scala:23]: No, it's not needed at the moment so I can remove it

File: app/config/FrontendAppConfig.scala (hunk 4 above)
Comment (amaalali) [reply app/config/FrontendAppConfig.scala:28]: No there is no requirements as yet

File: conf/app.routes (hunk 2 above)
Comment (ericohwotu) [reply conf/app.routes:9]: Okay thats fine then. if we need i think we can use the accessibility service?

Comment (amaalali) [reply conf/app.routes:9]: I'm not sure 🤔  Since those are available to the public we may need to host it on this service

Comment (ericohwotu) [reply conf/app.routes:9]: Ah i get you.. true.. good point

**Review by ericohwotu — APPROVED** [review]
